Zend/zend_compile.c

```diff
@@ -6467,10 +6467,13 @@ void zend_compile_func_decl(znode *result, zend_ast *ast, zend_bool toplevel) /*
 	uint32_t fallback_return_type = 0;
 
 	if (is_method) {
-		if (zend_string_equals_literal_ci(decl->name, "__construct")
-			|| zend_string_equals_literal_ci(decl->name, "__destruct")) {
-			fallback_return_type = IS_VOID;
-		} else if (zend_string_equals_literal_ci(decl->name, "__toString")) {
+		if (zend_is_constructor(decl->name)) {
+			zend_error(E_DEPRECATED,
```

Comment:
Very good approach! :)

Reply:
The patch itself isn't complete since now every time there is a constructor, a deprecation warning is thrown. WIP on fixing that.